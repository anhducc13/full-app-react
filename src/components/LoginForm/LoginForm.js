import React, { useState } from 'react';
import { setGlobal } from 'reactn';
import Avatar from '@material-ui/core/Avatar';
import CssBaseline from '@material-ui/core/CssBaseline';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import { Link as RouteLink } from 'react-router-dom';
import { Button as ButtonCustom } from '../shared/Button';
import { InputText } from '../shared/InputText';
import Swal from 'sweetalert2';
import { authService } from 'services';
import { validateUsername, validatePassword } from 'helpers/validators';


const useStyles = makeStyles(theme => ({
  '@global': {
    body: {
      backgroundColor: theme.palette.common.white,
    },
  },
  paper: {
    marginTop: theme.spacing(2),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
    width: 60,
    height: 60,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
}));

export const LoginForm = (props) => {
  const classes = useStyles();

  const initTextField = {
    value: '',
    error: '',
    valid: false,
  };

  const [usernameField, setUsernameField] = useState(initTextField);
  const [passwordField, setPasswordField] = useState(initTextField);

  const onChangeUsername = (e) => {
    let inputValidate = Object.assign(usernameField);
    inputValidate.value = e.target.value;
    const outputValidate = validateUsername(inputValidate);
    setUsernameField(outputValidate);
  }

  const onChangePassword = (e) => {
    let inputValidate = Object.assign(passwordField);
    inputValidate.value = e.target.value;
    const outputValidate = validatePassword(inputValidate);
    setPasswordField(outputValidate);
  }

  const onSubmitForm = (e) => {
    e.preventDefault();
    const params = {
      "username": usernameField.value,
      "password": passwordField.value
    }
    setGlobal({
      loading: true
    });
    authService.login(params)
      .then(res => {
        setGlobal({
          loading: false
        });
        if (res.status === 200) {
          const userInfoLogin = res.data;
          localStorage.setItem('userInfoLogin', JSON.stringify(userInfoLogin));
          Swal.fire(
            'Thành công!',
            `Xin chào ${userInfoLogin['username']} quay trở lại`,
            'success'
          )
            .then(() => {
              props.history.push('/home');
            })
        }
      })
      .catch(err => {
        console.log(err)
        setGlobal({
          loading: false
        });
        if (err.response && err.response.status === 404) {
          Swal.fire(
            'Có lỗi xảy ra!',
            'Tài khoản hoặc mật khẩu không chính xác',
            'error'
          )
        } else if (err.reponse && err.response.status === 400) {
          Swal.fire(
            'Có lỗi xảy ra!',
            'Dữ liệu đầu vào có vấn đề',
            'error'
          )
        } else {
          props.history.push('/error');
        }
      })
  }


  return (
    <Container component="main" maxWidth="sm">
      <CssBaseline />
      <div className={classes.paper}>
        <Avatar className={classes.avatar}>
          <LockOutlinedIcon />
        </Avatar>
        <Typography component="h1" variant="h4">
          Login
        </Typography>
        <form className={classes.form} onSubmit={onSubmitForm} noValidate>
          <InputText
            textField={usernameField}
            name="username"
            label="Username"
            type="text"
            onChange={onChangeUsername}
          />
          <InputText
            textField={passwordField}
            name="password"
            label="Password"
            type="password"
            onChange={onChangePassword}
          />
          <ButtonCustom
            displayText="Login"
            disabled={!usernameField.valid || !passwordField.valid}
          />
          <Grid container>
            <Grid item xs>
              <Link to="/forgot-password" variant="body2" component={RouteLink}>
                Forgot password?
              </Link>
            </Grid>
            <Grid item>
              <Link to="/register" variant="body2" component={RouteLink}>
                Don't have an account? Register
              </Link>
            </Grid>
          </Grid>
        </form>
      </div>
    </Container>
  );
}