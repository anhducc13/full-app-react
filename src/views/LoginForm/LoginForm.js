import React from 'react';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import { Link as RouteLink} from 'react-router-dom';


const useStyles = makeStyles(theme => ({
  '@global': {
    body: {
      backgroundColor: theme.palette.common.white,
    },
  },
  paper: {
    marginTop: theme.spacing(2),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
    width: 60,
    height: 60,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));

export const LoginForm = (props) => {
  const classes = useStyles();
  const { usernameField, passwordField } = props.textField;
  const { onChangeUsername, onChangePassword } = props.handleChange
  const submitForm = props.handleSubmit

  return (
    <Container component="main" maxWidth="sm">
      <CssBaseline />
      <div className={classes.paper}>
        <Avatar className={classes.avatar}>
          <LockOutlinedIcon />
        </Avatar>
        <Typography component="h1" variant="h4">
          Login
        </Typography>
        <form className={classes.form} onSubmit={submitForm} noValidate>
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            id="username"
            label="Username"
            name="username"
            autoFocus
            value={usernameField.value}
            error={usernameField.error !== ''}
            onChange={onChangeUsername}
            helperText={usernameField.valid ? "" : usernameField.error}
          />
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            name="password"
            label="Password"
            type="password"
            id="password"
            value={passwordField.value}
            error={passwordField.error !== ''}
            onChange={onChangePassword}
            helperText={passwordField.error ? "" : passwordField.error}
          />
          {/* <FormControlLabel
              control={<Checkbox value="remember" color="primary" />}
              label="Remember me"
            /> */}
          <Button
            size="large"
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
            disabled={!usernameField.valid || !passwordField.valid}
          >
            Sign In
          </Button>
          <Grid container>
            <Grid item xs>
              <Link to="/forgot-password" variant="body2" component={RouteLink}>
                Forgot password?
              </Link>
            </Grid>
            <Grid item>
              <Link to="/register" variant="body2" component={RouteLink}>
                {"Don't have an account? Register"}
              </Link>
            </Grid>
          </Grid>
        </form>
      </div>
    </Container>
  );
}