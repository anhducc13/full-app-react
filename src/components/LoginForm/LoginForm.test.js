import React from 'react';
import { mount, shallow } from 'enzyme';

import { LoginForm } from './LoginForm';


describe('Login Form Render', () => {
  it('should render correctly', () => {
    const component = shallow(<LoginForm />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});

// describe('Login form tests', () => {
  
// });