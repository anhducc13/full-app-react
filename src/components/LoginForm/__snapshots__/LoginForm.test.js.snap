// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form Render should render correctly 1`] = `
<WithStyles(ForwardRef(Container))
  component="main"
  maxWidth="sm"
>
  <CssBaseline />
  <div
    className="makeStyles-paper-1"
  >
    <WithStyles(ForwardRef(Avatar))
      className="makeStyles-avatar-2"
    >
      <LockOutlinedIcon />
    </WithStyles(ForwardRef(Avatar))>
    <WithStyles(ForwardRef(Typography))
      component="h1"
      variant="h4"
    >
      Login
    </WithStyles(ForwardRef(Typography))>
    <form
      className="makeStyles-form-3"
      noValidate={true}
      onSubmit={[Function]}
    >
      <InputText
        label="Username"
        name="username"
        onChange={[Function]}
        textField={
          Object {
            "error": "",
            "valid": false,
            "value": "",
          }
        }
        type="text"
      />
      <InputText
        label="Password"
        name="password"
        onChange={[Function]}
        textField={
          Object {
            "error": "",
            "valid": false,
            "value": "",
          }
        }
        type="password"
      />
      <Button
        disabled={true}
        displayText="Login"
      />
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
        >
          <WithStyles(ForwardRef(Link))
            component={[Function]}
            to="/forgot-password"
            variant="body2"
          >
            Forgot password?
          </WithStyles(ForwardRef(Link))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Link))
            component={[Function]}
            to="/register"
            variant="body2"
          >
            Don't have an account? Register
          </WithStyles(ForwardRef(Link))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </form>
  </div>
</WithStyles(ForwardRef(Container))>
`;
